src/Models/NonhydrostaticModels/boundary_mass_fluxes.jl

```diff
@@ -0,0 +1,238 @@
+using Oceananigans.BoundaryConditions: BoundaryCondition, Open, PerturbationAdvection, FlatExtrapolation
+using Oceananigans.AbstractOperations: Integral, Ax, Ay, Az, GridMetricOperation
+using Oceananigans.Fields: Field, interior, XFaceField, YFaceField, ZFaceField
+
+const OBC  = BoundaryCondition{<:Open} # OpenBoundaryCondition
+const IOBC = BoundaryCondition{<:Open{<:Nothing}} # "Imposed-velocity" OpenBoundaryCondition (with no matching scheme)
+const FIOBC = BoundaryCondition{<:Open{<:Nothing}, <:Number} # "Fixed-imposed-velocity" OpenBoundaryCondition
+const ZIOBC = BoundaryCondition{<:Open{<:Nothing}, <:Nothing} # "Zero-imposed-velocity" OpenBoundaryCondition (no-inflow)
+
+function get_west_area(grid)
+    dA = GridMetricOperation((Face, Center, Center), Ax, grid)
+    ∫dA = sum(dA, dims=(2, 3))
+    return interior(∫dA, 1, 1, 1) |> maximum
+end
+
+function get_east_area(grid)
+    dA = GridMetricOperation((Face, Center, Center), Ax, grid)
+    ∫dA = sum(dA, dims=(2, 3))
+    return interior(∫dA, grid.Nx+1, 1, 1) |> maximum
```

Comment:
Do we reduce with `maximum` just as a way of extracting a single value, or something more going on here?

Reply:
Just to extract a single value. I found `maximum` to be the simplest from all the options I tested. I can add a comment to make it clearer, or extract it another way.

Reply:
I guess just indexing into it might be easier to interpret --- `∫dA[grid.Nx+1, 1, 1]`

Reply:
`maximum` does more than just a scalar index, it will allocate memory for the result, compute a reduction, and then output the result using `@allowscalar` to transfer the data from CPU to GPU. If you simply index into the array with `@allowscalar` you avoid the extraneous operations and get the info you need directly.